content/docs/tasks/telemetry/export-logs-through-stackdriver.md

```diff
@@ -0,0 +1,301 @@
+weight: 70
+---
+
+This task shows how to export Istio Access Logs to different sinks like
+BigQuery, GCS, Pub/Sub through Stackdriver A user can do analytics on data
+through Istio using logs flowing from Istio to StackDriver and then exporting
+these logs to various configured sinks like BigQuery, GCS or Pub/Sub.
+
+The [Bookinfo](/docs/guides/bookinfo/) sample application is used as the example
+application throughout this task.
+
+## Before you begin
+
+[Install Istio](/docs/setup/) in your cluster and deploy an application.
+
+## Configuring Istio for exporting logs to various sinks
+
+Istio exports logs through logentry template configured for Mixer as [accesslog
+entry](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/config.yaml#L134:9).
+This specifies all the variables that would be available for analysis. It
+contains information like source service, destination service, auth
+metrics(coming..) among others. Following is a diagram of the pipeline:
+
+{{< image width="75%" ratio="69.52%"
+link="../img/istio-analytics-using-stackdriver.png" alt="Image is not available"
+title="Diagram of exporting logs from Istio to StackDriver for analysis" >}}
+
+Istio supports exporting logs to Stackdriver which can be extended to export
+logs to your favorite sink like BigQuery, Pub/Sub or GCS. Please follow steps
+below to setup your favorite sink for exporting logs first and then Stackdriver
+in Istio.
+
+### Setting up various log sinks
+
+Common setup for all sinks:
+
+1. Enable StackDriver Monitoring API for the project
+1. Make sure `principalEmail` that would be setting up the sink has write access
+to the project and Logging Admin role permissions.
+
+#### BigQuery
+
+1.  Create a BigQuery dataset where you would like logs to get exported in
+    BigQuery
+1.  Note down it’s id to be passed to Stackdriver handler as destination for the
+    sink. It would be of the form
+    `bigquery.googleapis.com/projects/[PROJECT_ID]/datasets/[DATASET_ID]`
+1.  Give sink’s writer identity: cloud-logs@system.gserviceaccount.com BigQuery
+    data editor role in IAM
+
+#### GCS
+
+1.  Create a GCS bucket where you would like logs to get exported in BigQuery.
+1.  Note down it’s id to be passed to Stackdriver handler as destination for the
+    sink. It would be of the form `storage.googleapis.com/[BUCKET_ID`
+1.  Give sink’s writer identity: cloud-logs@system.gserviceaccount.com storage
+    object creator role in IAM
+
+#### Pub/Sub
+
+1.  Create a topic where you would like logs to get exported in Pub/Sub
+1.  Note down it’s id to be passed to Stackdriver handler as destination for the
+    sink. It would be of the form
+    `pubsub.googleapis.com/projects/[PROJECT_ID]/topics/[TOPIC_ID]`
+1.  Give sink’s writer identity: cloud-logs@system.gserviceaccount.com Pub/Sub
+    Publisher role in IAM
+
+### Setting up Stackdriver
+
+To configure exporting logs to Stackdriver, you have to write a handler for
```

Comment:
Suggestion:

> A Stackdriver handler must be created to export data to Stackdriver. The configuration schema for a Stackdriver handler can be found [here]. 

I'd also link `handler` to the topic discussing what a handler is.

Reply:
Fixed